code/android-lifecycle-library/src/main/java/com/adobe/marketing/mobile/lifecycle/LifecycleUtil.java

```diff
@@ -57,13 +62,56 @@ private static String dateToISO8601String(final Date timestamp, final String tim
 	}
 
 	/**
-	 * Formats the locale value from SystemInfoService and replaces '_' with '-'
+	 * Formats the locale value by replacing '_' with '-'. Uses {@link Locale#toString()} to retrieve
+	 * the language tag.
+	 * <p>
+	 * Note. the use of {@code Locale#toString()} does not return a value formatted to BCP 47.
+	 * For example, script codes are appended to the locale as "-#scriptCode", as in "zh-HK-#Hant",
+	 * where BCP 47 requires the format as "zh-Hant-HK".
 	 *
+	 * @see #formatLocaleXDM(Locale) 
 	 * @param locale active locale value
 	 * @return string representation of the locale
 	 */
 	static String formatLocale(final Locale locale) {
 		return locale == null ? null : locale.toString().replace('_', '-');
 	}
 
+	/**
+	 * Format the locale to the string format used in XDM.
+	 * For Android API version >= Lollipop (21), returns {@link Locale#toLanguageTag()}.
+	 * For Android API version < 21, returns a concatenation of {@link Locale#getLanguage()}
+	 * and {@link Locale#getCountry()}, separated by '-'.
+	 *
+	 * @param locale active Locale value
+	 * @return String representation of the locale
+	 */
+	@SuppressLint("NewApi")
+	static String formatLocaleXDM(final Locale locale) {
```

Comment:
Changes look good overall, but I would not limit this API to the xdm field only but standardizing it in the SDK usages. Can we make this a new deviceInfo API `getActiveLocaleBCPString()` and remove any custom formatters in extensions code?

Reply:
I like this idea as it provides consistent locale usage across the entire SDK. @praveek what do you think?

Reply:
Yes. It sounds good as other extensions can reuse the implementation from DeviceInforming.

Reply:
The current thinking is to change the logic in DeviceInforming.getLocaleString() to return a BCP-47 style string instead of adding a second API. We will need to vet this change with some stakeholders which consume the user-agent and accept-language headers as those formates will changes slightly. I'll take up this change in another PR.